tests/kafkatest/services/copycat.py

```diff
@@ -77,7 +78,52 @@ def clean_node(self, node):
         node.account.ssh("rm -rf /mnt/copycat.pid /mnt/copycat.log /mnt/copycat.properties  " + " ".join(self.config_filenames() + self.files), allow_fail=False)
 
     def config_filenames(self):
-        return ["/mnt/copycat-connector-" + str(idx) + ".properties" for idx, template in enumerate(self.connector_config_templates)]
+        return ["/mnt/copycat-connector-" + str(idx) + ".properties" for idx, template in enumerate(self.connector_config_templates or [])]
+
+
+    def list_connectors(self, node=None):
+        return self._rest('/connectors', node=node)
+
+    def create_connector(self, config, node=None):
+        create_request = {
+            'name': config['name'],
+            'config': config
+        }
+        return self._rest('/connectors', create_request, node=node, method="POST")
+
+    def get_connector(self, name, node=None):
+        return self._rest('/connectors/' + name, node=node)
+
+    def get_connector_config(self, name, node=None):
+        return self._rest('/connectors/' + name + '/config', node=node)
+
+    def set_connector_config(self, name, config, node=None):
+        return self._rest('/connectors/' + name + '/config', config, node=node, method="PUT")
+
+    def get_connector_tasks(self, name, node=None):
+        return self._rest('/connectors/' + name + '/tasks', node=node)
+
+    def delete_connector(self, name, node=None):
+        return self._rest('/connectors/' + name, node=node, method="DELETE")
+
+    def _rest(self, path, body=None, node=None, method="GET"):
+        if node is None:
+            node = random.choice(self.nodes)
+
+        meth = getattr(requests, method.lower())
+        url = self._base_url(node) + path
+        resp = meth(url, json=body)
+        self.logger.debug("%s %s response: %d", url, method, resp.status_code)
+        if resp.status_code > 400:
+            raise CopycatRestError(resp.status_code, resp.text, resp.url)
+        if resp.status_code == 204:
+            return None
+        else:
+            return resp.json()
+
+
+    def _base_url(self, node):
+        return 'http://' + node.account.hostname + ':' + '8083'
```

Comment:
Thats awesome. I really like that our tests simply call the REST interface from Python.